File: BackofficeSpringboot/src/main/java/itu/mbds/tpt/entity/Echange.java
package itu.mbds.tpt.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "\"Echanges\"")
public class Echange {

    @Id
    @Column(name = "\"id\"")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "\"responder_id\"", nullable = false)
    private Utilisateur responder;

    @Column(name = "\"description\"")
    private String description;

    @Column(name = "\"created_at\"", nullable = false, updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private LocalDateTime createdAt;

    @Column(name = "\"updated_at\"", nullable = false)
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "\"post_id\"", nullable = false)
    private Post post;

    @Column(name = "\"status\"", length = 20, nullable = false)
    private String status;

    @OneToMany(mappedBy = "echange", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<EchangeDetail> echangeDetails;
}

File: ClientWinForm/Models/Echange.cs
using System;
using System.Collections.Generic;

namespace ClientWinForm.Models
{
    public class Echange
    {
        public int Id { get; set; }
        public DateTime created_at { get; set; }
        public string Description { get; set; }
        public string Status { get; set; }
        public DateTime updated_at { get; set; }
        public int post_id { get; set; }
        public int proposer_id { get; set; }
        public int responder_id { get; set; }
        public Utilisateur Proposer { get; set; }
        public Utilisateur Responder { get; set; }
        public Post Poste { get; set; }
        public List<EchangeDetail> EchangeDetails { get; set; }
    }

    public class EchangeDetail
    {
        public int Id { get; set; }
        public int EchangeId { get; set; }
        public int ObjetId { get; set; }
        public Objet Objet { get; set; }
    }
}
